File: src/java/com/springbok/pattern/PatternSRR_403V01Test.java
/* Copyright (C) 2016 Springbok LLC <http://springbok.io/> All rights reserved. */
package com.springbok.pattern;

import com.springbok.utility.TestUtility;
import org.junit.Test;

import java.util.HashMap;
import java.util.Map;

import static org.junit.Assert.assertTrue;

/**
 * Tests methods of PatternSRR_403V01 class.
 */
public class PatternSRR_403V01Test {
	private static final double Phi0_input = 2;
	private static final double GainMax_input = 57;

	private static final double[] Phi_input = new double[] { 0.2, 2, 5, 150 };
	private static final double[] G_expected = new double[] { 56.880000000000003, 45.000000000000000,
			29.551499783199059, 0 };
	private static final double[] Gx_expected = new double[] { 26.879999999999999, 24.000000000000000,
			9.956349637772750, 0 };

	/**
	 * Tests PatternSRR_403V01 method.
	 */
	@Test
	public void test_PatternSRR_403V01() {
		PatternSRR_403V01 p = new PatternSRR_403V01(this.Phi0_input);

		assertTrue(TestUtility.isDoublesEquals(2, p.getPhi0()));
	}

	/**
	 * Tests gain method.
	 */
	@Test
	public void test_gain() {
		PatternSRR_403V01 p = new PatternSRR_403V01(this.Phi0_input);

		Gain[] gains = new Gain[Phi_input.length];
		Map<String, Object> options = new HashMap<String, Object>();
		options.put("DoValidate", false);
		options.put("GainMax", GainMax_input);

		for (int i = 0; i < gains.length; i++) {
			Gain g = p.gain(Phi_input[i], options);
			gains[i] = g;
		}
		for (int i = 0; i < gains.length; i++) {
			assertTrue(TestUtility.isDoublesEquals(G_expected[i], gains[i].G));
			assertTrue(TestUtility.isDoublesEquals(Gx_expected[i], gains[i].Gx));
		}
	}
}

File: src/matlab/com/springbok/pattern/PatternSRR_403V01Test.m
classdef PatternSRR_403V01Test < TestUtility
% Tests methods of PatternSRR_403V01 class.
  
% Copyright (C) 2016 Springbok LLC <http://springbok.io/> All rights reserved.

  properties (Constant = true)
    
    Phi0_input = 2;
    GainMax_input = 57;
    
    self = struct( ...
        'Phi0', 2);
    
    Phi_input = [0.2; 2; 5; 150];
    
    G_expected = [ ...
        56.880000000000003
        45.000000000000000
        29.551499783199059
        0
                 ];
    Gx_expected = [ ...
        26.879999999999999
        24.000000000000000
        9.956349637772750
        0
                  ];
    
  end % properties (Constant = true)
  
  methods
    
    function this = PatternSRR_403V01Test(logFId, testMode)
    % Constructs a PatternSRR_403V01Test
    %
    % Parameters:
    %   logFId - Log file identifier
    %   testMode - Test mode, if 'interactive' then beeps and pauses
      
    % Invoke superclass constructor
      if nargin == 0
        superArgs = {};
        
      else
        superArgs{1} = logFId;
        superArgs{2} = testMode;
        
      end % if
      this@TestUtility(superArgs{:});
      
    end % PatternSRR_403V01Test()
    
    function test_PatternSRR_403V01(this)
    % Tests PatternSRR_403V01 method.
      
      p = PatternSRR_403V01(this.Phi0_input);
      
      this.assert_true( ...
          'PatternSRR_403V01', ...
          'PatternSRR_403V01', ...
          this.HIGH_PRECISION_DESC, ...
          PatternUtility.compare_pattern(this.self, p));
      
    end % test_PatternSRR_403V01()
    
    function test_gain(this)
    % Tests gain method.
      
      p = PatternSRR_403V01(this.Phi0_input);
      
      [G_actual, Gx_actual] = p.gain(this.Phi_input, 'GainMax', this.GainMax_input);
      
      t(1) = max(abs(G_actual - this.G_expected) < this.HIGH_PRECISION);
      t(2) = max(abs(Gx_actual - this.Gx_expected) < this.HIGH_PRECISION);
      
      this.assert_true( ...
          'PatternSRR_403V01', ...
          'gain', ...
          this.HIGH_PRECISION_DESC, ...
          min(t));
      
      [G, Gx, fH] = p.gain([0:.001:180], 'GainMax', this.GainMax_input, 'PlotFlag', true);
      pause(1); close(fH);
      
      [G, Gx, fH] = p.gain([0:.001:180], 'PlotFlag', true);
      pause(1); close(fH);
      
    end % test_gain()
    
  end % methods
  
end % classdef
